containers/EmojiPicker.tsx

```diff
@@ -108,49 +136,82 @@ export const EmojiPicker = () => {
       } else {
         addReactionToMessage(currentUser, message, emoji);
       }
-      drawerRef?.current?.closeDrawer(closeMenu);
+      bottomSheetRef?.current?.dismiss();
+      closeMenu();
     },
-    [closeMenu, conversation, currentUser, currentUserEmojiMap, message]
+    [
+      conversation,
+      currentUser,
+      currentUserEmojiMap,
+      message,
+      bottomSheetRef,
+      closeMenu,
+    ]
+  );
+
+  const debouncedFilter = useMemo(
+    () =>
+      debounce((value: string) => {
+        setFilteredReactions(filterEmojis(value));
+        setHasInput(value.length > 0);
+      }, 150),
+    []
+  );
+
+  const onTextInputChange = useCallback(
+    (value: string) => {
+      debouncedFilter(value);
+    },
+    [debouncedFilter]
   );
 
   return (
-    <Drawer visible={visible} onClose={closeMenu} ref={drawerRef} showHandle>
-      <EmojiSearchBar value={searchInput} setValue={setSearchInput} />
-      <View style={styles.container}>
-        {searchInput.length > 0 ? (
+    <BottomSheetModal
+      onClose={closeMenu}
+      ref={bottomSheetRef}
+      enableDynamicSizing
+      topInset={insets.top}
+      snapPoints={["40%", "100%"]}
+    >
+      <BottomSheetContentContainer>
+        <BottomSheetHeader title={translate("choose_reaction")} hasClose />
+        <TextField
+          ref={textInputRef}
+          onChangeText={onTextInputChange}
+          placeholder={translate("search_emojis")}
+          clearButtonMode="always"
+          containerStyle={$inputContainer}
+        />
+      </BottomSheetContentContainer>
+
+      <View style={themed($container)}>
+        {hasInput ? (
           <EmojiRowList emojis={filteredReactions} onPress={handleReaction} />
         ) : (
-          <>
-            <EmojiRowList
-              emojis={categorizedEmojis}
-              onPress={handleReaction}
-              ListHeader={
-                <>
-                  {/* Removing until customization is ready */}
-                  {/* <Text style={styles.headerText}>Your Reactions</Text>
-                  <EmojiRowList emojis={myEmojis} onPress={handleReaction} /> */}
-                  <Text style={styles.headerText}>All</Text>
-                </>
-              }
-            />
-          </>
+          <EmojiRowList
+            emojis={categorizedEmojis}
+            onPress={handleReaction}
+            ListHeader={
+              <Text preset="smaller" style={themed($headerText)}>
+                {translate("emoji_picker_all")}
+              </Text>
+            }
+          />
```

Comment:
I think rendering both absolute positioning and animating them with reanimated might be more performant